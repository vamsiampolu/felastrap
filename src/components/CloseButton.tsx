import type { ComponentPropsWithoutRef } from "react";
import { useFela } from "react-fela";
import closeIcon from "/close-button.svg";
import { StyleProps, Theme } from "./common";

const close = () => {
  const borderRadius = {
    borderTopLeftRadius: "0.375rem",
    borderTopRightRadius: "0.375rem",
    borderBottomRightRadius: "0.375rem",
    borderBottomLeftRadius: "0.375rem",
  };

  return {
    ...borderRadius,
    borderColor: "rgb(0,0,0)",
    backgroundColor: "transparent",
    backgroundImage: `url(${closeIcon})`,
    backgroundPositionX: "center",
    backgroundOrigin: "padding-box",
    backgroundClip: "border-box",
    backgroundPositionY: "1em",
    backgroundRepeat: "no-repeat",
    backgroundSize: "auto",
    boxSizing: "content-box",
    border: 0,
    color: "#000",
    height: "1em",
    opacity: 0.5,
    paddingTop: "0.25em",
    paddingBottom: "0.25em",
    paddingLeft: "0.25em",
    paddingRight: "0.25em",
    width: "1em",
    ":focus": {
      boxShadow: "0 0 0 0.25rem rgba(13, 110, 253, 0.25)",
      opacity: 1,
      outline: 0,
    },
    ":disabled": {
      opacity: 0.25,
      pointerEvents: "none",
      userSelect: "none",
    },
  };
};

const alertRule = ({ alert }: StyleProps<CloseButtonProps>) => {
  return alert
    ? {
        paddingTop: "1.25rem",
        paddingBottom: "1.25rem",
        paddingLeft: "1rem",
        paddingRight: "1rem",
        position: "absolute",
        right: 0,
        top: 0,
        zIndex: 2,
      }
    : {};
};

const modalRule = ({ modal }: StyleProps<CloseButtonProps>) =>
  modal
    ? {
        padding: "0.5rem 0.5rem",
        margin: "-0.5rem -0.5rem -0.5rem auto",
      }
    : {};

export type CloseButtonProps = ComponentPropsWithoutRef<"button"> &
  Partial<{ alert?: boolean; modal?: boolean }>;

export const CloseButton = ({
  alert = false,
  modal = false,
  className = "",
  ...other
}: CloseButtonProps) => {
  const { css } = useFela<Theme, CloseButtonProps>({ alert, modal });
  const cx = `${css([
    close,
    alertRule,
    modalRule,
  ])} close-button ${className}`.trim();
  return (
    <button
      data-testid="close-button"
      className={cx}
      aria-label="Close"
      {...other}
    />
  );
};
