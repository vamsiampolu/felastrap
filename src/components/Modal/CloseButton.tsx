import { ComponentPropsWithoutRef } from "react";
import { useFela } from "react-fela";
import close from "/close-button.svg";

const rule = () => ({
  marginTop: "-2px",
  appearance: "none",
  textAlign: "end",
  cursor: "pointer",
  border: 0,
  position: "absolute",
  right: "2rem",
  top: "1rem",
  padding: "1rem",
  backgroundColor: "transparent",
  backgroundImage: `url(${close})`,
  backgroundRepeat: "no-repeat",
  backgroundSize: "16px",
  backgroundPosition: "center",
});

export type CloseButtonProps = Omit<
  ComponentPropsWithoutRef<"button">,
  "type" | "aria-label"
>;

export const CloseButton = ({
  onClick,
  className = "",
  ...props
}: CloseButtonProps) => {
  const { css } = useFela();
  const cx = `${css(rule)} bs-close ${className}`.trim();
  return (
    <button
      data-testid="modal-close-button"
      className={cx}
      onClick={onClick}
      type="button"
      aria-label="Close"
      {...props}
    ></button>
  );
};
