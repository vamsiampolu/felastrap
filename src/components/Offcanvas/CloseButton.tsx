import { useFela } from "react-fela";
import closeIcon from "/close-button.svg";
import { ComponentPropsWithoutRef } from "react";

const borderRadius = {
  borderTopLeftRadius: "0.375rem",
  borderTopRightRadius: "0.375rem",
  borderBottomRightRadius: "0.375rem",
  borderBottomLeftRadius: "0.375rem",
};

const rule = () => ({
  ...borderRadius,
  borderColor: "rgb(0,0,0)",
  backgroundColor: "transparent",
  backgroundImage: `url(${closeIcon})`,
  backgroundPositionX: "center",
  backgroundOrigin: "padding-box",
  backgroundClip: "border-box",
  backgroundPositionY: "1em",
  backgroundRepeat: "no-repeat",
  backgroundSize: "auto",
  boxSizing: "content-box",
  border: 0,
  color: "#000",
  height: "1em",
  opacity: 0.5,
  paddingTop: "10px",
  paddingBottom: "10px",
  paddingLeft: "0.25em",
  paddingRight: "0.25em",
  width: "1em",
  ":focus": {
    boxShadow: "0 0 0 0.25rem rgba(13, 110, 253, 0.25)",
    opacity: 1,
    outline: 0,
  },
  ":disabled": {
    opacity: 0.25,
    pointerEvents: "none",
    userSelect: "none",
  },
});

const offcanvasRule = () => ({
  padding: "calc(1rem * 0.5) calc(1rem * 0.5)",
  marginTop: "calc(-0.5 * 1rem)",
  marginRight: "calc(-0.5 * 1rem)",
  marginBottom: "calc(-0.5 * 1rem)",
});

export type CloseButtonProps = Omit<
  ComponentPropsWithoutRef<"button">,
  "aria-label" | "type" | "children"
>;

export const CloseButton = ({ className = "", ...props }: CloseButtonProps) => {
  const { css } = useFela();
  const cx = `${css([
    rule,
    offcanvasRule,
  ])} offcanvas-btn-close ${className}`.trim();
  return (
    <button
      data-testid="offcanvas-close-button"
      type="button"
      className={cx}
      aria-label="Close"
      {...props}
    />
  );
};
