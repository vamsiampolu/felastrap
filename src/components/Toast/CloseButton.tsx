import { useFela } from "react-fela";
import closeIcon from "/close-button.svg";
import { ComponentPropsWithoutRef } from "react";

const borderRadius = {
  borderTopLeftRadius: "0.375rem",
  borderTopRightRadius: "0.375rem",
  borderBottomRightRadius: "0.375rem",
  borderBottomLeftRadius: "0.375rem",
};

const rule = () => {
  return {
    ...borderRadius,
    marginRight: "calc(-0.5 * 0.75rem)",
    marginLeft: "0.75rem",
    paddingTop: "10px !important",
    paddingBottom: "10px !important",
    borderColor: "rgb(0,0,0)",
    backgroundColor: "transparent",
    backgroundImage: `url(${closeIcon})`,
    backgroundPositionX: "center",
    backgroundOrigin: "padding-box",
    backgroundClip: "border-box",
    backgroundPositionY: "1em",
    backgroundRepeat: "no-repeat",
    backgroundSize: "auto",
    boxSizing: "content-box",
    border: 0,
    color: "#000",
    height: "1em",
    opacity: 0.5,
    width: "1em",
    ":focus": {
      boxShadow: "0 0 0 0.25rem rgba(13, 110, 253, 0.25)",
      opacity: 1,
      outline: 0,
    },
    ":disabled": {
      opacity: 0.25,
      pointerEvents: "none",
      userSelect: "none",
    },
  };
};

export type CloseButtonProps = Omit<
  ComponentPropsWithoutRef<"button">,
  "aria-label" | "type" | "children"
>;
export const CloseButton = ({ className = "", ...props }: CloseButtonProps) => {
  const { css } = useFela();
  const cx = `${css(rule)} toast-btn-close ${className}`;
  return (
    <button
      data-testid="toast-close-button"
      type="button"
      className={cx}
      aria-label="Close"
      {...props}
    />
  );
};
