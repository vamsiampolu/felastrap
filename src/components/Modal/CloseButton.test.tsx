import { CloseButton } from "./CloseButton";
import { IRenderer } from "fela";
import {
  Wrapper,
  createRenderer,
  createSnapshot,
  cssReboot,
} from "../../test-helpers";
import { render, screen } from "@testing-library/react";

describe("components/Modal/CloseButton", () => {
  let renderer: IRenderer;

  beforeEach(() => {
    renderer = createRenderer();
    cssReboot(renderer);
  });

  afterEach(() => {
    renderer.clear();
  });

  it("should render a Modal CloseButton", async () => {
    const component = <CloseButton onClick={() => undefined} />;
    render(<Wrapper renderer={renderer}>{component}</Wrapper>);

    expect(screen.getByTestId("modal-close-button")).toBeInTheDocument();
    expect(screen.queryByTestId("modal-close-button")).toHaveAttribute(
      "aria-label",
      "Close",
    );

    const snapshot = await createSnapshot({ component, renderer });
    expect(snapshot).toMatchSnapshot();
  });
});
