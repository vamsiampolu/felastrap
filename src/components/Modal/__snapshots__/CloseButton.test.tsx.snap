// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Modal/CloseButton should render a Modal CloseButton 1`] = `
".a {
  margin-top: -2px;
}
.b {
  appearance: none;
}
.c {
  text-align: end;
}
.d {
  cursor: pointer;
}
.e {
  position: absolute;
}
.f {
  right: 2rem;
}
.g {
  top: 1rem;
}
.h {
  background-color: transparent;
}
.i {
  background-image: url(test-file-stub);
}
.j {
  background-repeat: no-repeat;
}
.k {
  background-size: 16px;
}
.l {
  background-position: center;
}
.m {
  padding-top: 1rem;
}
.n {
  padding-right: 1rem;
}
.o {
  padding-bottom: 1rem;
}
.p {
  padding-left: 1rem;
}
.q {
  border-top-width: 0;
}
.r {
  border-right-width: 0;
}
.s {
  border-bottom-width: 0;
}
.t {
  border-left-width: 0;
}


<button
  data-testid="modal-close-button"
  class="a b c d e f g h i j k l m n o p q r s t bs-close"
  type="button"
  aria-label="Close"
></button
>;
"
`;
